zone/bonuses.cpp

```diff
@@ -146,7 +146,16 @@ void Mob::CalcItemBonuses(StatBonuses* b) {
 	int16 i;
 
 	for (i = EQ::invslot::BONUS_BEGIN; i <= EQ::invslot::BONUS_SKILL_END; i++) {
-		const EQ::ItemInstance* inst = GetInv().GetItem(i);
+		const EQ::ItemInstance* inst = nullptr;
+
+		if (IsOfClientBot()) {
+			inst = GetInv().GetItem(i);
+		} else if (IsMerc()) {
+			inst = database.CreateItem(CastToMerc()->GetEquipmentItemID(i));
```

Comment:
We need an alternative to this; this re-introduces the memory leak again. I'm happy to pair on this if we need to figure out an alternative.

https://github.com/EQEmu/Server/pull/3331

https://github.com/EQEmu/Server/pull/3331/files#diff-66d4c1caa8c324b219b395e271593e026eaf6baf97851378de2ba8a8f145d102L158

See heap trace (CreateItem)

![image](https://github.com/EQEmu/Server/assets/3319450/984f5809-ba11-4462-8edf-c28a52aa6785)